### ISSUE TITLE
Minified react error #31 when editing dashboard panel with Google Cloud Monitoring data source

### ISSUE THREAD

**Issue by reidlevesque** [grafana/grafana#48907]
**What happened**:

When I edit a panel in a dashboard where the data source is [Google Cloud Monitoring](https://grafana.com/docs/grafana/latest/datasources/google-cloud-monitoring/). I get the following error:

```
Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7B%7D for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

**What you expected to happen**:

I expect to be able to use the [query editor](https://grafana.com/docs/grafana/latest/datasources/google-cloud-monitoring/#using-the-query-editor) to create a query. 

**How to reproduce it (as minimally and precisely as possible)**:

1. Configure a data source using Google Cloud Monitoring using `GCE Default Service Account` authentication (see image)
2. `Save & test` and ensure you get the green check mark
3. Create new dashboard and select `Add a new panel`.
4. Select the `Google Cloud Monitoring` data source. This may work the first time. If it does, select another data source and then re-select `Google Cloud Monitoring`

<img width="826" alt="image" src="https://user-images.githubusercontent.com/13453006/167875025-07c6e300-857c-43b6-8fab-1eaaa577b9ea.png">

**Anything else we need to know?**:

* [Issue #47872](https://github.com/grafana/grafana/issues/47872) seemed related but upgrading to 8.5.0 did not resolve the issue.
* The plugin does work when I import the dashboards that come with the data source (i.e. the dashboards show data). I am able to reproduce the issue with those dashboards by editing a panel, switching the data source to something that's not Google Cloud Monitoring and then switching back.

**Environment**:
- Grafana version: 8.5.2 (also tested on 8.5.0, 8.4.6, 8.4.4)
- Data source type & version: [Google Cloud Monitoring](https://grafana.com/docs/grafana/latest/datasources/google-cloud-monitoring/) 1.0.0
- OS Grafana is installed on: [grafana/grafana:8.5.2](https://hub.docker.com/layers/grafana/grafana/grafana/8.5.2/images/sha256-60666139f672e0ff3d71395102a813e7ff85e8afd7cc5c2d4676abcb00a357a4?context=explore) docker image running in GKE
- User OS & Browser: MacOS, Firefox 100.0 (also tested on Safari 15.4, Brave 1.38.111)
- Grafana plugins: -
- Others: -

**Comment by zuchka** [comment]
do you think you could try and replicate this using this dummy dashboard on our public sandbox?

https://play.grafana.org/d/cvaNLTX7k/grafana-grafana-48907?orgId=1&editPanel=2

This datasource's configuration is testing green. It is also using Google JWT file / GCE default service account for auth

**Comment by reidlevesque** [comment]
I am not able to reproduce the issue on that dashboard. We have been able to work around the issue by using `Google JWT File` authentication; the problem does not happen when using that authentication method. It only happens when we use `GCE Default Service Account` authentication. I was not able to see which authentication method is used for the Google Cloud Monitoring datasource in your public sandbox. Could you confirm the public sandbox is using `GCE Default Service Account` authentication?

**Comment by zuchka** [comment]
I checked and we are indeed using `JWT`. So this will need some deeper research it looks like 👍

**Comment by aangelisc** [comment]
Hi @reidlevesque, I can confirm that I've managed to reproduce this issue in GCP. The error seems to be related to the `Project` component. I'll continue to investigate and keep this issue updated.

**Comment by aangelisc** [comment]
Hi @reidlevesque, #49510 has been merged which should fix this issue. It'll be released in the next patch version of 8.5 😊
